src/main/scala/better/files/package.scala

```diff
@@ -46,27 +49,45 @@ package object files {
 
     def createIfNotExists(): File = if (javaFile.exists()) this else Files.createFile(javaPath)
 
-    def appendLines(lines: String*): File = Files.write(javaPath, lines, defaultCharset(), StandardOpenOption.APPEND, StandardOpenOption.CREATE)
-    def <<(line: String): File = appendLines(line)
-    val >>: = << _
+    def content(implicit codec: Codec): BufferedSource = Source.fromFile(javaFile)(codec)
+    def source(implicit codec: Codec): BufferedSource = content(codec)
+
+    def bytes: Iterator[Byte] = {
+      val stream = in
+      Iterator.continually(stream.read()).takeWhile(-1 !=).map(_.toByte)  //TODO: close the stream
+    }
+
+    def chars(implicit codec: Codec): Iterator[Char] = content(codec)
+
+    def lines(implicit codec: Codec): Iterator[String] = content(codec).getLines()
+
+    def contentAsString(implicit codec: Codec): String = new String(bytes.toArray, codec)
+    def `!`(implicit codec: Codec): String = contentAsString(codec)
+
+    def appendLines(lines: String*)(implicit codec: Codec): File = Files.write(javaPath, lines, codec, StandardOpenOption.APPEND, StandardOpenOption.CREATE)
+    def <<(line: String)(implicit codec: Codec): File = appendLines(line)(codec)
+    def >>:(line: String)(implicit codec: Codec): File = appendLines(line)(codec)
     def appendNewLine: File = appendLines("")
 
-    def write(bytes: File.Contents): File = Files.write(javaPath, bytes)
-    def write(text: String): File = Files.write(javaPath, text)
-    val overwrite = write _       //TODO: Method alias macro
-    val < = write _
-    val `>:` = write _
+    def write(bytes: Iterator[Byte]): File = Files.write(javaPath, bytes.toArray) //TODO: Large I/O?
+
+    def write(text: String)(implicit codec: Codec): File = write(text.getBytes(codec).toIterator)
+    def overwrite(text: String)(implicit codec: Codec) = write(text)(codec)
+    def <(text: String)(implicit codec: Codec) = write(text)(codec)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [ >:  is a reserved keyword don't use it as a name.](https://www.codacy.com/app/pathikrit/better-files/file/977849228/issues/source?bid=2860326&fileBranchId=2864209#l77)